### PR TITLE
Add Rated self-report

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/config/settings.py [hunk 1]
```diff
@@ -215,6 +218,16 @@ def set(
...
         self.skip_startup_checks = decouple_config('SKIP_STARTUP_CHECKS', default=False, cast=bool)
 
+        # Rated API
+        self.rated_api_url = decouple_config('RATED_API_URL', default='https://api.rated.network')
+
+        # StakeWise API URL
+        self.stakewise_api_url = (
+            'https://mainnet-api.stakewise.io/graphql'
+            if self.network == MAINNET
+            else 'https://holesky-api.stakewise.io/graphql'
```
Comment (cyc60) [inline src/config/settings.py:228]: what about other networks? Mb move this into `networks.py`

File: src/config/settings.py [hunk 2]
```diff
@@ -215,6 +218,16 @@ def set(
 
         self.skip_startup_checks = decouple_config('SKIP_STARTUP_CHECKS', default=False, cast=bool)
 
+        # Rated API
+        self.rated_api_url = decouple_config('RATED_API_URL', default='https://api.rated.network')
```
Comment (cyc60) [inline src/config/settings.py:222]: also can be moved to `networks.py`?

File: src/commands/rated_self_report.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            (
+                f'Failed to report validators. Status code: {response.status_code}, '
+                f'Response: {response.text}'
+            ),
+            bold=True,
+            fg='red',
+        )
+
+
+def fetch_validators(vault_address: str, api_url: str) -> list:
```
Comment (cyc60) [inline src/commands/rated_self_report.py:112]: Use gql lib for graph query

File: src/commands/rated_self_report.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    response = requests.post(
+        f'{settings.rated_api_url}/v0/selfReports/validators',
+        headers=headers,
+        data=json.dumps(payload),
+        timeout=15,
+    )
+
+    if response.status_code == 201:
+        click.secho('Successfully reported validators.', bold=True, fg='green')
+        click.secho(f'{response.json}')
```
Comment (cyc60) [inline src/commands/rated_self_report.py:100]:
```suggestion
        click.secho(response.json)
```

Comment (cyc60) [reply src/commands/rated_self_report.py:100]: or remove this line

File: src/common/graph.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    @retry_gql_errors(delay=DEFAULT_RETRY_TIME)
+    async def run_query(self, query: DocumentNode, params: dict | None = None) -> dict:
+        result = await self.gql_client.execute_async(query, variable_values=params)
+        return result
+
+    async def get_vault_validators(self, vault: str) -> list[str]:
+        query = gql(
+            """
+            query Validators($vaultAddress: String!) {
+              vaultValidators(
```
Comment (evgeny-stakewise) [inline src/common/graph.py:30]: There is a limit on number of validators in a single query. See `first` param. It is 100 by default. Need to use pagination.

File: src/commands/rated_self_report.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    settings.set(
+        vault=vault,
+        vault_dir=vault_config.vault_dir,
+        network=network,
+        execution_endpoints='',
+        consensus_endpoints='',
+    )
+    click.secho('Rated self report')
+
+    graph_client = GraphClient()
```
Comment (cyc60) [inline src/commands/rated_self_report.py:72]: can be moved into `_report_validators` func

File: src/commands/rated_self_report.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': f'Bearer {token}',
+            'X-Rated-Network': network,
+        }
+
+        response = requests.post(
+            f'{settings.network_config.RATED_API_URL}/v0/selfReports/validators',
+            headers=headers,
+            data=json.dumps(payload),
```
Comment (cyc60) [inline src/commands/rated_self_report.py:103]:
```suggestion
            json=payload,
```

File: src/common/graph.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                vaultAddress: $vaultAddress
+                statusIn: "active_ongoing"
```
[the remark this answers is not in the document]
Comment (evgeny-stakewise) [reply src/common/graph.py:34]: No changes, no comments.

File: src/commands/rated_self_report.py [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+)
+@click.command(help='Self-report your validators to the Rated Network.')
+def rated_self_report(
+    vault: HexAddress,
+    network: str,
+    pool_tag: str,
+    token: str,
+    data_dir: str,
+) -> None:
+    if network == 'gnosis':
```
Comment (cyc60) [inline src/commands/rated_self_report.py:59]: better to check if networks in supported networks. That about chiado foe example

File: src/commands/rated_self_report.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+    vault_config.load()
+
+    settings.set(
+        vault=vault,
+        vault_dir=vault_config.vault_dir,
+        network=network,
+        execution_endpoints='',
+        consensus_endpoints='',
+    )
+    click.secho('Rated self report')
```
Comment (cyc60) [inline src/commands/rated_self_report.py:73]:
```suggestion
    click.secho('Starting rated self report...')
```

**Review by cyc60 — APPROVED** [review]

**Review by cyc60 — APPROVED** [review]

**Pull request merged**